File: desktop/src/at/fhooe/mc/emg/tools/conconi/ConconiData.java
package at.fhooe.mc.emg.tools.conconi;

import at.fhooe.mc.emg.client.ChannelData;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * Author:  Mescht
 * Date:    08.07.2017
 */
class ConconiData implements Serializable {

    private List<ChannelData> roundData;

    ConconiData() {
        roundData = new ArrayList<>();
    }

    void addRoundData(ChannelData crd) {
        roundData.add(crd);
    }

    ChannelData getRoundData(int index) {
        return roundData.get(index);
    }

    int getRoundCount() {
        return roundData.size();
    }

}

File: core/src/at/fhooe/mc/emg/core/tools/conconi/ConconiData.kt
package at.fhooe.mc.emg.core.tools.conconi

import at.fhooe.mc.emg.clientdriver.model.EmgData
import java.io.Serializable
import java.util.*

/**
 * Author:  Mescht
 * Date:    08.07.2017
 */
internal class ConconiData : Serializable {

    private val roundData: MutableList<EmgData>

    val roundCount: Int
        get() = roundData.size

    init {
        roundData = ArrayList()
    }

    fun addRoundData(crd: EmgData) {
        roundData.add(crd)
    }

    fun getRoundData(index: Int): EmgData {
        return roundData[index]
    }

}
